samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst

```diff
@@ -0,0 +1,351 @@
+
+.. _CONFIG_CLOUD_CONNECTION_RETRY_TIMEOUT_SECONDS:
+
+CONFIG_CLOUD_CONNECTION_RETRY_TIMEOUT_SECONDS - Cloud connection retry timeout (seconds)
+   If connecting to `nRF Cloud`_ takes longer than this timeout, it will be reattempted.
+
+.. _CONFIG_CLOUD_CONNECTION_REESTABLISH_DELAY_SECONDS:
+
+CONFIG_CLOUD_CONNECTION_REESTABLISH_DELAY_SECONDS - Cloud connection reestablishment delay (seconds)
+   After the connection to `nRF Cloud`_ has been reset, wait this amount of time before attempting to connect again.
+
+.. _CONFIG_CLOUD_READY_TIMEOUT_SECONDS:
+
+CONFIG_CLOUD_READY_TIMEOUT_SECONDS - Cloud readiness timeout (seconds)
+   If the connection to `nRF Cloud`_ does not become ready within this timeout, the sample will reset its connection and try again.
+
+.. _CONFIG_DATE_TIME_ESTABLISHMENT_TIMEOUT_SECONDS:
+
+CONFIG_DATE_TIME_ESTABLISHMENT_TIMEOUT_SECONDS - Modem date and time establishment timeout (seconds)
+   The sample will wait this many seconds for the modem to determine the current date and time before giving up and moving on.
+
+.. _CONFIG_APPLICATION_THREAD_STACK_SIZE:
+
+CONFIG_APPLICATION_THREAD_STACK_SIZE - Application Thread Stack Size (bytes)
+   Sets the stack size (in bytes) for the application thread of the sample.
+
+.. _CONFIG_CONNECTION_THREAD_STACK_SIZE:
+
+CONFIG_CONNECTION_THREAD_STACK_SIZE - Connection Thread Stack Size (bytes)
+   Sets the stack size (in bytes) for the connection thread of the sample.
+
+.. _CONFIG_MESSAGE_THREAD_STACK_SIZE:
+
+CONFIG_MESSAGE_THREAD_STACK_SIZE - Message Queue Thread Stack Size (bytes)
+   Sets the stack size (in bytes) for the message queue processing thread of the sample.
+
+.. _CONFIG_MAX_OUTGOING_MESSAGES:
+
+CONFIG_MAX_OUTGOING_MESSAGES - Outgoing message maximum
+   Sets the maximum number of messages that can be in the outgoing message queue.
+   Messages submitted past this limit will not be enqueued.
+
+.. _CONFIG_MAX_CONSECUTIVE_SEND_FAILURES:
+
+CONFIG_MAX_CONSECUTIVE_SEND_FAILURES - Max outgoing consecutive send failures
+   Sets the maximum number of device messages which may fail to send before a connection reset and cooldown is triggered.
+
+.. _CONFIG_CONSECUTIVE_SEND_FAILURE_COOLDOWN_SECONDS:
+
+CONFIG_CONSECUTIVE_SEND_FAILURE_COOLDOWN_SECONDS - Cooldown after max consecutive send failures exceeded (seconds)
+   If a connection reset is triggered by too many failed device messages, the sample will wait this long (in seconds) before trying again.
+
+.. _CONFIG_SENSOR_SAMPLE_INTERVAL_SECONDS:
+
+CONFIG_SENSOR_SAMPLE_INTERVAL_SECONDS - Sensor sampling interval (seconds)
+   Sets the time to wait between each temperature sensor sample.
+
+.. warning::
+   Every time a sensor sample is sent, it will interrupt any attempted GNSS fix.
+   If you set this interval too short, there will not be enough time to ever acquire a fix.
+   In good conditions, and with AGPS data, one minute is generally just barely long enough to get a fix most of the time.
```

Comment:
I don't think we want make such statement and it should not take so long anyway with A-GPS. I did not get into reviewing the code yet - maybe there is something else taking up the time if you experienced it takes such a long time?

Reply:
Did you use Thingy for testing btw? The TTFF (time to first fix) will be somewhat longer on Thingy due to worse internal antenna. I guess my point is just that TTFF will vary wildly depending on a million different things and I am hesitant to put out a number (let alone one that will make the GNSS performance look bad). YMMV

Reply:
Since writing that statement my fix times appear to be on the order of 10 to 20 seconds, but a sampling interval of 60 seconds with 20 seconds of extra time for cellular fallback results in a 40 second GNSS fix timeout, which is twice the nominal delay under ideal conditions (20 seconds), so I felt that one minute was the shortest interval that one can **_safely_** pick, including all tolerancing.

I could probably soften the languaging quite a bit -- the way I wrote it implies that anything under a minute will start failing to fix a significant portion of the time, and that's not quite true either.

Do you think that:

> In good conditions, and with AGPS data, one minute is a reasonable interval for reliably getting a fix.

Oh, and yes, I **_am_** using a thingy!

Reply:
So it turns out that I had a bit of a brain glitch and documented this issue **_twice_**, once in the sensor sampling period section, and once in the location sampling period section. The location sampling period, AFAICT, does not actually come into play for ensuring GNSS fix. I have combined both of my statements on this issue into a single statement under the CONFIG_SENSOR_SAMPLE_INTERVAL_SECONDS Kconfig option documentation.

Reply:
I am fine with your proposal. Maybe something you wrote here could also be included. Fore example: "In good conditions, and with A-GPS data, one minute is a reasonable interval for reliably getting a location estimate. This allows using long enough value for GNSS_FIX_TIMEOUT_SECONDS, while still leaving enough time for falling back to cellular positioning in case GNSS fails to obtain a fix." 

Btw. the code is very well written and easy to read, good job!

Reply:
Thank you! Code readability was probably my #1 goal, so it is a relief to hear that it is readable to someone other than myself.